stream.go

```diff
@@ -13,6 +13,63 @@ import (
 	"github.com/gorilla/websocket"
 )
 
+// Error codes
+const (
+	CloseUnauthorized        = 4001
+	CloseBadRequest          = 4002
+	CloseInsufficientCredits = 4003
+	CloseServerShuttingDown  = 4010
+	CloseNoInstanceAvailable = 4013
+	CloseTooManyRequests     = 4029
+)
+
+// Whether or not connection should be retried
+var shouldErrorRetry = map[int]bool{
+	CloseUnauthorized:        false,
+	CloseBadRequest:          false,
+	CloseInsufficientCredits: false,
+	CloseServerShuttingDown:  true,
+	CloseNoInstanceAvailable: true,
+	CloseTooManyRequests:     false,
+}
+
+// Whether or not connection should be retried
+var errorMsgs = map[int]string{
+	CloseUnauthorized:        "Unauthorized. The provided access token is invalid.",
+	CloseBadRequest:          "Bad request. The connection’s content-type is invalid, metadata contains too many characters or the custom vocabulary does not exist with that id.",
+	CloseInsufficientCredits: "Insufficient credits. The client does not have enough credits to continue the streaming session.",
+	CloseServerShuttingDown:  "Server shutting down. The connection was terminated due to the server shutting down.",
+	CloseNoInstanceAvailable: "No instance available. No available streaming instances were found. User should attempt to retry the connection later.",
+	CloseTooManyRequests:     "Too many requests. The number of concurrent connections exceeded the limit. Contact customer support to increase it.",
+}
+
+// A close message from rev see https://www.rev.ai/docs/streaming#section/Error-Codes
+type RevError struct {
+	// Error code
+	Code int
+
+	// The error string
+	Text string
+
+	// Whether or not the connection should be retried
+	ShouldRetry bool
+}
+
+func (e RevError) Error() string {
+	return fmt.Sprintf("Streaming error: %s", e.Text)
+}
+
+// Check if the code is a Rev error if so return it.
+func IsRevError(code int) (bool, error) {
+	errorString, exists := errorMsgs[code]
+	if exists {
+		shouldRetry := shouldErrorRetry[code]
```

Comment:
@threeaccents some errors are recoverably by dialing again. at the moment I'm thinking leave this up to the user. Do you have thoughts.